index.js

```diff
@@ -30,12 +31,19 @@ var zlib = require('zlib')
 module.exports = compression
 module.exports.filter = shouldCompress
 
+/**
+ * @const
+ * whether current node version has brotli support
+ */
+var hasBrotliSupport = 'createBrotliCompress' in zlib
+
 /**
  * Module variables.
  * @private
  */
-
 var cacheControlNoTransformRegExp = /(?:^|,)\s*?no-transform\s*?(?:,|$)/
+var SUPPORTED_ENCODING = hasBrotliSupport ? ['br', 'gzip', 'deflate', 'identity'] : ['gzip', 'deflate', 'identity']
+var PREFERRED_ENCODING = hasBrotliSupport ? ['br', 'gzip'] : ['gzip']
```

Comment:
This is a non-blocking nitpick, if others disagree we can disregard. It is my personal opinion that when the constant we define is only used one place in the file, all we are doing is forcing folks to scroll back up in a file to connect what is going on on line 199. This doesn't help keep things cleaner and is an abstraction that actually *adds* cognitive load when reading the code. 

IMO it would be better to do this in-line this back into 199.

Reply:
I asked for it to be done this way, and disagree with the overhead. The variables are well named, and having it inline makes it harder to read and worse for performance.

Reply:
I agree, keeping it inline makes it harder to read